File: IS-A.java
//Implementing Is-A Relationship in Java


class Shape {
    void draw() {
        System.out.println("Drawing a shape.");
    }
}
class Circle extends Shape {
    void drawCircle() {
        System.out.println("Drawing a circle.");
    }
}
public class Main {
    public static void main(String[] args) {
        Circle myCircle = new Circle();
        myCircle.draw();
        myCircle.drawCircle();
    }
}

File: IS-A.js
//Implementing Is-A Relationship in JavaScript


function Shape() {}
Shape.prototype.draw = function() {
    console.log("Drawing a shape.");
};
function Circle() {
    Shape.call(this);
}
Circle.prototype = Object.create(Shape.prototype);
Circle.prototype.constructor = Circle;
Circle.prototype.drawCircle = function() {
    console.log("Drawing a circle.");
};
var myCircle = new Circle();
myCircle.draw();
myCircle.drawCircle();